scripts/relup-base-packages.sh

```diff
@@ -63,8 +63,21 @@ cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")/.."
 mkdir -p _upgrade_base
 pushd _upgrade_base
 
+# For 4.5+, we upgrade from OTP 24.1.5-3 to 24.3.4.2-1, so we must manually
+# check the old OTP releases.
+otp_vsn_for() {
+  case "${1#[e|v]}" in
+    4.4.*)
+      echo "24.1.5-3"
+      ;;
+    4.5.*)
+      echo "24.3.4.2-1"
+      ;;
+  esac
+}
+
 for tag in $(../scripts/relup-base-vsns.sh $EDITION | xargs echo -n); do
-    filename="$PROFILE-${tag#[e|v]}-otp$OTP_VSN-$SYSTEM-$ARCH.zip"
+    filename="$PROFILE-${tag#[e|v]}-otp$(otp_vsn_for "$tag")-$SYSTEM-$ARCH.zip"
```

Comment:
we no longer need line 31?

Reply:
Oops, better use the script for getting the OTP of 4.5+ than hard-coding the version.

Reply:
I think it's fine to hard code here because we can not change the past.

Reply:
Sure, the old versions are indeed [hardcoded](https://github.com/emqx/emqx/pull/8684/files#diff-f42fb6f8358f71f02d16a191bb1e9a672e62a1a668fa2ef1cd9706806a758fe4R71), but I meant that the current one (for 4.5) could use the [current OTP](https://github.com/emqx/emqx/pull/8684/files#diff-f42fb6f8358f71f02d16a191bb1e9a672e62a1a668fa2ef1cd9706806a758fe4R74) from the script.  Does it make sense?